delphi/__main__.py

```diff
@@ -135,7 +135,7 @@ async def process_cache(
         max_examples=latent_cfg.max_examples,
     )
     sampler = partial(sample, cfg=experiment_cfg)
-    loader = LatentLoader(dataset, constructor=constructor, sampler=sampler)
+    dataset.build_iterator(constructor, sampler, None)
```

Comment:
Functional code that returns something is almost always better then code that mutates things and doesn't return anything